# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/build/XPCOMInit.cpp
# Commit: 36b0646e8649
# Full Hash: 36b0646e86498662e770756ad54b1fd648a82184
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-04-07 21:44:02
# Regressor Bug: 1620594
# File Overlap Count: 1
# Description:
#   Bug 1620594 - Part 7: Remove TabGroup and SystemGroup. r=nika,bas
#   
#   TabGroup never really made any difference in which thread something go
#   dispatched to. This was the intended use, but development of TabGroups
#   with abstract main threads never made it that far. The good thing is
# ==============================================================================

diff -r a5ad3b179c92 -r 36b0646e8649 xpcom/build/XPCOMInit.cpp
--- a/xpcom/build/XPCOMInit.cpp	Tue Apr 07 15:17:14 2020 +0000
+++ b/xpcom/build/XPCOMInit.cpp	Tue Apr 07 15:17:47 2020 +0000
@@ -71,7 +71,6 @@
 #include "mozilla/Services.h"
 #include "mozilla/Omnijar.h"
 #include "mozilla/ScriptPreloader.h"
-#include "mozilla/SystemGroup.h"
 #include "mozilla/Telemetry.h"
 #include "mozilla/BackgroundHangMonitor.h"
 
@@ -328,9 +327,6 @@
   }
   AUTO_PROFILER_INIT2;
 
-  // Init the mozilla::SystemGroup for dispatching main thread runnables.
-  mozilla::SystemGroup::InitStatic();
-
   // Set up the timer globals/timer thread
   rv = nsTimerImpl::Startup();
   if (NS_WARN_IF(NS_FAILED(rv))) {
@@ -515,9 +511,6 @@
     return rv;
   }
 
-  // Init the mozilla::SystemGroup for dispatching main thread runnables.
-  mozilla::SystemGroup::InitStatic();
-
   // Set up the timer globals/timer thread.
   rv = nsTimerImpl::Startup();
   if (NS_WARN_IF(NS_FAILED(rv))) {
@@ -781,9 +774,6 @@
   nsComponentManagerImpl::gComponentManager = nullptr;
   nsCategoryManager::Destroy();
 
-  // Shut down SystemGroup for main thread dispatching.
-  SystemGroup::Shutdown();
-
   GkRust_Shutdown();
 
 #ifdef NS_FREE_PERMANENT_DATA